File: j2k/testData/fileOrElement/typeCasting/IntCompatibleInBitwiseOps.java
public class TestIntCompatibleInBitwiseOps {
    public int testByte(byte b) {
        return b & 0x0F | 0xF0;
    }

    public int testShort(short s) {
        return s & 0x0F | 0xF0;
    }

    public int testInt(int i) {
        return i & 0x0F | 0xF0;
    }
}

File: j2k/testData/fileOrElement/typeCasting/IntCompatibleInBitwiseOps.kt
class TestIntCompatibleInBitwiseOps {
    fun testByte(b: Byte): Int {
        return (b and 0x0F or (0xF0).toByte()).toInt()
    }

    fun testShort(s: Short): Int {
        return (s and 0x0F or 0xF0).toInt()
    }

    fun testInt(i: Int): Int {
        return i and 0x0F or 0xF0
    }
}